### PR TITLE
Fix #6137: Do not show reader mode button if content fails to parse.

### LINKED ISSUE #6137: Hackerone 1727527
See https://hackerone.com/reports/1727527

### PR DESCRIPTION
**Pull request by iccub**
## Summary of Changes

This pull request fixes #6137 

## Submitter Checklist:

- [x] *Unit Tests* are updated to cover new or changed functionality
- [x] User-facing strings use `NSLocalizableString()`
- [x] New or updated UI has been tested across:
  - [x] Light & dark mode
  - [x] Different size classes (iPhone, landscape, iPad)
  - [x] Different dynamic type sizes

## Test Plan:

STR in hackerone ticket description

## Reviewer Checklist:

- [x] Issues include necessary QA labels:
  - `QA/(Yes|No)`
  - `bug` / `enhancement`
- [x] Necessary [security reviews](https://github.com/brave/security/issues/new/choose) have taken place.
- [x] Adequate unit test coverage exists to prevent regressions.
- [x] Adequate test plan exists for QA to validate (if applicable).
- [x] Issue and pull request is assigned to a milestone (should happen at merge time).

### CODE REVIEW

File: Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/ReaderModeScriptHandler.swift [hunk 1]
```diff
@@ -185,6 +185,10 @@ struct ReadabilityResult {
       if let credits = dict["byline"] as? String {
         self.credits = credits
       }
+      
+      if !JSONSerialization.isValidJSONObject(dict) {
+        return nil
+      }
```
Comment (kylehickinson) [inline Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/ReaderModeScriptHandler.swift]: Wouldn't it be better to have this at the top?

File: Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/ReaderModeScriptHandler.swift [hunk 2]
```diff
@@ -167,6 +167,10 @@ struct ReadabilityResult {
   var credits = ""
 
   init?(object: AnyObject?) {
+    guard JSONSerialization.isValidJSONObject(dict) { else
+      return nil
+    }
```
Comment (kylehickinson) [inline Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/ReaderModeScriptHandler.swift]: Too high now, `dict` isn't defined until 1 line down 😅

Comment (iccub) [reply Client/Frontend/UserContent/UserScripts/Scripts_Dynamic/ScriptHandlers/Sandboxed/ReaderModeScriptHandler.swift]: lol, wanted to fix during the meeting call [*] thanks for the catch

**Review by kylehickinson — APPROVED** [review]

**Review by stoletheminerals — APPROVED** [review]
sgtm

**Pull request merged**